pkg/join/join.go

```diff
@@ -0,0 +1,554 @@
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/datafile"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/brokersecret"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/servicediscoverycr"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinercr"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinerop"
+	"github.com/submariner-io/submariner-operator/pkg/version"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type Options struct {
+	ClusterID                     string
+	ServiceCIDR                   string
+	ClusterCIDR                   string
+	GlobalnetCIDR                 string
+	Repository                    string
+	ImageVersion                  string
+	NattPort                      int
+	IkePort                       int
+	PreferredServer               bool
+	ForceUDPEncaps                bool
+	ColorCodes                    string
+	NatTraversal                  bool
+	IgnoreRequirements            bool
+	GlobalnetEnabled              bool
+	IpsecDebug                    bool
+	SubmarinerDebug               bool
+	OperatorDebug                 bool
+	LabelGateway                  bool
+	LoadBalancerEnabled           bool
+	CableDriver                   string
+	GlobalnetClusterSize          uint
+	CustomDomains                 []string
+	ImageOverrideArr              []string
+	HealthCheckEnable             bool
+	HealthCheckInterval           uint64
+	HealthCheckMaxPacketLossCount uint64
+	CorednsCustomConfigMap        string
+}
+
+var status = cli.NewStatus()
+
+func SubmarinerCluster(subctlData *datafile.SubctlData, jo Options) error{
+	// Missing information
+	qs := []*survey.Question{}
+
+	determineClusterID(jo)
+
+	if valid, err := isValidClusterID(jo.ClusterID); !valid {
+		fmt.Printf("Error: %s\n", err.Error())
+
+		qs = append(qs, &survey.Question{
+			Name:   "clusterID",
+			Prompt: &survey.Input{Message: "What is your cluster ID?"},
+			Validate: func(val interface{}) error {
+				str, ok := val.(string)
+				if !ok {
+					return nil
+				}
+				_, err := isValidClusterID(str)
+				return err
+			},
+		})
+	}
+
+	if jo.ColorCodes == "" {
+		qs = append(qs, &survey.Question{
+			Name:     "colorCodes",
+			Prompt:   &survey.Input{Message: "What color codes should be used (e.g. \"blue\")?"},
+			Validate: survey.Required,
+		})
+	}
+
+	if len(qs) > 0 {
+		answers := struct {
+			ClusterID  string
+			ColorCodes string
+		}{}
+
+		err := survey.Ask(qs, &answers)
+		// Most likely a programming error
+		utils.PanicOnError(err)
+
+		if len(answers.ClusterID) > 0 {
+			jo.ClusterID = answers.ClusterID
+		}
+
+		if len(answers.ColorCodes) > 0 {
+			jo.ColorCodes = answers.ColorCodes
+		}
+	}
```

Comment:
L90-132 are related to the CLI and should be in _cmd/subctl/join.go_. We shouldn't do any prompting here.